Documentation/guides/BindingJavaSource.md

```diff
@@ -0,0 +1,106 @@
+---
+id:
+title: "Binding Java Source"
+dateupdated: 2021-05-21
+---
+
+# Overview
+
+With .net 6 Bindings can be done on not only .jar and .aar files
+but also raw .java code. This will allow developers of bindings
+or applications to write custom API's for underlying bindings
+and expose them easily to the developer.
+
+Applications already had the ability to include `AndroidJavaSource`
+items in them, these would then be compiled into the final `classes.dex`.
+However this custom java source code was not "bound" and if users wanted
+to call these classes they would need to do that manually.
+
+The `AndroidJavaSource` Item Group now supports a `Bind` attribute. This
+will instruct the build system to not only compile the code but also
+produce a C# API for it.
+
+```xml
+<ItemGroup>
+    <AndroidJavaSource Include="MyClass.java" Bind="True" />
+</ItemGroup>
+```
+
+The `Bind` attribute will default to `true` when for any `AndroidJavaSource`
+item. If you do not want to bind the java code set `Bind` to
+`false`.
+
+```xml
+<ItemGroup>
+    <AndroidJavaSource Include="MyClass.java" Bind="False" />
+</ItemGroup>
+```
+
+# Example
+
+Consider the following Java code.
+
+```java
+package com.xamarin.test;
+
+class MyClass {
+    public boolean IsDave (string name)
+    {
+        return name.equals ("Dave");
+    }
+}
+```
+
+We want to bind this into a C# API in our app project.
+So we can add a new `AndroidJavaSource` items with the `Bind`
+attribute set to `true`.
+
+```xml
+<ItemGroup>
+    <AndroidJavaSource Include="MyClass.java" Bind="True" />
```

Comment:
Since `%(Bind)` is True, do we actually need to set it?

Additionally, since this is now for .NET 6+, shouldn't there be "default item group" support for…something, removing the need for this item group *entirely*?

Related: https://github.com/xamarin/xamarin-android/pull/5926#discussion_r638077508